pkg/cluster/provider.go

```diff
@@ -69,12 +70,20 @@ func NewProvider(options ...ProviderOption) *Provider {
 		}
 	}
 
+	isLegitProvider := func(prog string) bool {
+		out, err := exec.Command(prog, "-v").Output()
+		if err != nil {
+			return false
+		}
+		return strings.Contains(string(out), prog)
```

Comment:
does this actually work? `podman --version` spits out $0

Reply:
per @amwat, so if a user did the symlink trick to avoid installing a package, I'm not sure this works.

Reply:
```sh
more docker
#!/bin/sh

exec /usr/bin/podman "$@"

 aojea  /  usr  bin  sudo ./docker -v
podman version 1.9.1
```

Reply:
i mean if you put a symlink in path like `/usr/bin/docker -> /usr/bin/podman` ?

Reply:
```
root@instance-4:/home/amwat/amwat/kind# ls -al /usr/bin/docker
lrwxrwxrwx 1 root root 15 May 18 23:02 /usr/bin/docker -> /usr/bin/podman
root@instance-4:/home/amwat/amwat/kind# which docker
/usr/bin/docker
root@instance-4:/home/amwat/amwat/kind# docker --version
docker version 1.8.0
root@instance-4:/home/amwat/amwat/kind# podman --version
podman version 1.8.0
```

Reply:
also on my machine (macOS):
```
$ docker -v
Docker version 19.03.8, build afacb8b
```
note the capital D

Reply:
you are dight, for reference
```
 aojea  /  usr  bin  ./podman -v
podman version 1.9.1
 aojea  /  usr  bin  ./docker -v
docker version 1.9.1
 aojea  /  usr  bin  ls -al docker
lrwxrwxrwx. 1 root root 6 may 19 18:49 docker -> podman
```